DateRangePicker (#25)
mui-org/core-plus-pickers: docs/src/pages/components/date-range-picker/date-range-picker.md

> <PageMeta component="DateRangePicker" />

Author: Should I add the same for other components?

> <PageMeta component="DateRangePicker" />

Reviewer: We will need to but it relies on a more recent version of the documentation so we have no feedback that it's done right or wrong.

# Date range picker

<p class="description">Date pickers let the user select a range of dates.</p>

> ⚠️⚠️ The date range picker is unstable, it's **not suitable** for usage in production. ⚠️⚠️

> The date range picker will be made available in the coming months for production use as part of a paid extension to the community edition (MIT license) of Material-UI.
> This paid extension will include advanced components (rich data grid, date range picker, tree view drag & drop, etc.). Pricing for early access will start with an affordable plan.

The [date range pickers](https://material.io/components/pickers/) let users select a range of dates.

<Ad />

## Basic usage

> Basic DateRangePicker example, make sure that you can pass almost any prop of [DatePicker]('/api/date-range-picker')

Author: This mean actually date picker.

---
title: Date Range Picker React component
components: DateRangePicker
githubLabel: component: DateRangePicker
packageName: @material-ui/lab
materialDesign: https://material.io/components/date-pickers
---

# Date range picker

<p class="description">Date pickers let the user select a range of dates.</p>

> > ⚠️⚠️ The date range picker is unstable, it's **not suitable** for usage in production. ⚠️⚠️

Reviewer 2: ```suggestion
> ⚠️⚠️ The date range picker is unstable, and **not suitable** for usage in production. ⚠️⚠️
```

> > The date range picker will be made available in the coming months for production use as part of a paid extension to the community edition (MIT license) of Material-UI.

Reviewer 2: ```suggestion
> The date range picker will be made available in the coming months for production use as part of a paid extension (commercial license) to the community edition (MIT license) of Material-UI.
```
```suggestion
> The date range picker will be made available in the coming months for production use as part of a paid extension to the community edition (MIT license) of Material-UI.
```

> > This paid extension will include advanced components (rich data grid, date range picker, tree view drag & drop, etc.). Pricing for early access will start with an affordable plan.

Reviewer 2: ```suggestion
> This paid extension will include advanced components (rich data grid, date range picker, tree view drag & drop, etc.). [Early access](https://material-ui.com/store/items/material-ui-x/) starts at an affordable price.
```

(Are we saying that the final price will be unaffordable? 😄 )

> The [date range pickers](https://material.io/components/pickers/) let users select a range of dates.

Reviewer 2: ```suggestion
The [date range pickers](https://material.io/components/pickers/) let the user select a range of dates.
```

## Requirements

This component relies on the date management library of your choice. We support [date-fns](https://date-fns.org/), [luxon](https://moment.github.io/luxon/), [dayjs](https://github.com/iamkun/dayjs), [moment](https://momentjs.com/) and any other library via [date-io](https://github.com/dmtrKovalenko/date-io) adapter.

Please install any of these libraries and set up the right date engine by wrapping your root (or the highest level you wish the pickers to be available) with `LocalizationProvider`:

```jsx
// or @material-ui/lab/dateAdapter/{dayjs,luxon,moment} or any valid date-io adapter
import DateFnsAdapter from '@material-ui/lab/dateAdapter/date-fns';
import LocalizationProvider from '@material-ui/lab/LocalizationProvider';

function App() {
  return (
    <LocalizationProvider dateAdapter={DateFnsAdapter}>
      ...
    </LocalizationProvider>
  );
}
```

## Basic usage

> Basic DateRangePicker example, make sure that you can pass almost any prop of [DatePicker]('/api/date-picker/')

Reviewer 2: ```suggestion
Basic DateRangePicker example. Note that you can pass almost any prop from [DatePicker]('/api/date-picker/').
```

{{"demo": "pages/components/date-range-picker/BasicDateRangePicker.js"}}

## Responsiveness

> The date range picker component is designed and optimized for the device it runs on.

Reviewer 2: ```suggestion
The date range picker component is designed to be optimized for the device it runs on.
```

- The "Mobile" version works best for touch devices and small screens.
- The "Desktop" version works best for mouse devices and large screens.

By default, the `DateRangePicker` component uses a `@media (pointer: fine)` media query to determine which version to use.

> This can be customized by `desktopModeMediaQuery` prop.

Reviewer 2: ```suggestion
This can be customized with the `desktopModeMediaQuery` prop.
```

{{"demo": "pages/components/date-range-picker/ResponsiveDateRangePicker.js"}}

> ## Different amount of calendars

Reviewer 2: ```suggestion
## Different number of months
```

> Make sure that `calendars` prop is working only for desktop mode.

Reviewer 2: ```suggestion
Note that the `calendars` prop only works in desktop mode.
```
(We probably need to rethink the name of this prop...)

{{"demo": "pages/components/date-range-picker/CalendarsDateRangePicker.js"}}

## Disabling dates

> Disabling dates performs just like in simple `DatePicker`

Reviewer 2: ```suggestion
Disabling dates behaves the same as the simple `DatePicker`.
```

{{"demo": "pages/components/date-range-picker/MinMaxDateRangePicker.js"}}

## Custom input component

> You can customize rendering input by `renderInput` prop. For `DateRangePicker` it takes **2** parameters – for start and end input respectively.

Reviewer 2: ```suggestion
You can customize  the rendered input with the `renderInput` prop. For `DateRangePicker` it takes **2** parameters – for start and end input respectively.
```

> So if you need to render custom input make sure you will spread `ref` and `inputProps` correctly to the input components.

Reviewer 2: ```suggestion
If you need to render custom input make sure to spread `ref` and `inputProps` correctly to the input components.
```

{{"demo": "pages/components/date-range-picker/CustomDateRangeInputs.js"}}

## Static mode

> It is possible to render any picker without modal or popper. For that use `StaticDateRangePicker`.

Reviewer 2: ```suggestion
It is possible to render any picker without a modal or popper. For this use `StaticDateRangePicker`.
```
